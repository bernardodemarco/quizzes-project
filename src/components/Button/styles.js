import styled, { css } from 'styled-components';

export const Button = styled.button`
  ${({ theme, background, fontSize, fontWeight, lineHeight }) => css`
    background-color: ${background ? theme.colors.primary : theme.colors.white};
    color: ${background ? theme.colors.white : theme.colors.dark};
    font-size: ${theme.fonts.fontSizes.buttons[fontSize]};
    font-weight: ${theme.fonts.fontWeights[fontWeight]};
    line-height: ${lineHeight};

    font-family: inherit;
    width: 100%;
    height: 5.6rem;
    text-align: center;
    border: none;
    border-radius: 1.6rem;
    padding: 1.6rem 3.2rem;

    &:hover {
      cursor: pointer;
    }
  `}
`;
