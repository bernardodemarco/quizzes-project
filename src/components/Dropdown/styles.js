import styled, { css } from 'styled-components';

export const Dropdown = styled.ul`
  ${({ theme, isVisible }) => css`
    visibility: hidden;
    opacity: 0;

    transition: all 300ms ease-in-out;

    ${
      isVisible &&
      css`
      visibility: visible;
      opacity: 1;
    `
    }

    position: absolute;
    top: 10rem;
    right: 38%;

    width: 14.5rem;
    height: 15.2rem;

    background-color: ${theme.colors.white};
    border-radius: 0.8rem;
    box-shadow: 0rem 0.4rem 0.4rem rgba(0, 0, 0, 0.25);

    list-style: none;

    @media (max-width: 785px) {
      visibility: hidden;
      opacity: 0;
    }
  `}
`;

export const DropdownItem = styled.li`
  ${({ theme }) => css`
    width: 100%;

    padding: 1rem 4.5rem;

    border-bottom: 0.1rem solid ${theme.colors.gray};

    &:last-child {
      border-bottom: none;
    }

    &:hover {
      cursor: pointer;
    }
  `}
`;

export const Text = styled.p`
  ${({ theme }) => css`
    color: #000000;

    text-align: center;

    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    font-weight: ${theme.fonts.fontWeights.normal};
    line-height: 1.8rem;
  `}
`;
