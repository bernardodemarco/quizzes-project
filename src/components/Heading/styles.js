import styled, { css } from 'styled-components';

export const Heading = styled.h1`
  ${({ theme, fontSize, fontWeight, lineHeight }) => css`
    color: ${theme.colors.dark};
    font-size: ${theme.fonts.fontSizes.headings[fontSize]};
    font-weight: ${theme.fonts.fontWeights[fontWeight]};
    line-height: ${lineHeight};
  `}
`;
