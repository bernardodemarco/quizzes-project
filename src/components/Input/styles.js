import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    border: 0.1rem solid ${theme.colors.gray};
    border-radius: 1.2rem;

    width: 100%;
    height: 5.3rem;
    padding: 1.6rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
  `}
`;

export const Input = styled.input`
  ${({ theme }) => css`
    width: 75%;
    border: none;
    font-family: inherit;
    font-weight: ${theme.fonts.fontWeights.light};
    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    color: ${theme.colors.darkGray};
  `}
`;

export const Icon = styled.img`
  ${({ hover = false }) => css`
    width: 3rem;
    height: 2.5rem;

    &:hover {
      cursor: ${hover ? 'pointer' : 'default'};
    }
  `}
`;
