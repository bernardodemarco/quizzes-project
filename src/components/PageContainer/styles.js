import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    width: 100%;
    max-width: 114.6rem;
    min-height: 100vh;

    margin: 0 auto;
    overflow-x: hidden;

    display: flex;
    align-items: center;
  `}
`;
