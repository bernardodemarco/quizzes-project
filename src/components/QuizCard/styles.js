import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    overflow: hidden;

    border: 0.1rem solid ${theme.colors.gray};
    border-radius: 0.8rem;
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));

    width: 55.2rem;
    height: fit-content;
    background-color: ${theme.colors.white};

    &:hover {
      cursor: pointer;
    }

    @media ${theme.media.largeDevices} {
      width: 45rem;
    }

    @media ${theme.media.mediumDevices} {
      width: 55.2rem;
    }

    @media ${theme.media.smallDevices} {
      width: 90vw;

      p {
        text-align: start;
      }
    }
  `}
`;

export const Image = styled.div`
  ${({ imgSrc }) => css`
    // position set to relative in order to position <QuizDifficulty /> accordingly
    position: relative;

    width: 100%;
    height: 19.4rem;

    background-image: url(${imgSrc});
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
  `}
`;

export const TextContainer = styled.div`
  ${() => css`
    width: 100%;
    padding: 1.6rem 1.6rem 0.8rem 1.6rem;

    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 0.4rem;
  `}
`;

export const FeedbackContainer = styled.div`
  ${() => css`
    width: 100%;

    display: flex;
    justify-content: space-between;
  `}
`;
