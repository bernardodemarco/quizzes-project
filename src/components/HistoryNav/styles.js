import styled, { css } from 'styled-components';

import { Container as PageContainer } from './../PageContainer/styles';
import { ImageContainer as ReturnButton } from './../ReturnButton/styles';
import { Heading } from './../Heading/styles';

export const Nav = styled.nav`
  ${({ theme }) => css`
    z-index: 1;
    position: fixed;
    top: 0;
    right: 0;
    left: 0;

    height: 12rem;
    border-bottom: 0.1rem solid ${theme.colors.gray};

    ${PageContainer} {
      min-height: initial;
      height: 100%;
      justify-content: flex-start;
      align-items: center;
    }

    ${Heading} {
      z-index: 1;
      position: fixed;
      left: 14.7rem;
    }

    @media ${theme.media.mediumDevices} {
      ${ReturnButton} {
        left: 5rem;
      }
    }

    @media ${theme.media.smallDevices} {
      ${Heading} {
        left: 12rem;
        font-size: ${theme.fonts.fontSizes.headings.medium};
      }
    }
  `}
`;
