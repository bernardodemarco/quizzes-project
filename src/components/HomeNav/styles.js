import styled, { css } from 'styled-components';

import { Container as PageContainer } from '../PageContainer/styles';
import { Input, Container as InputContainer } from '../Input/styles';
import { Heading } from '../Heading/styles';

export const Button = styled.button`
  ${({ theme }) => css`
    display: none;

    z-index: 2;
    position: fixed;
    top: 3rem;
    right: 3rem;

    height: 4rem;
    width: 4rem;

    color: ${theme.colors.primary};
    background-color: ${theme.colors.lightGray};
    border: none;

    &:hover {
      cursor: pointer;
    }

    @media (max-width: 785px) {
      display: block;
    }
  `}
`;

export const Container = styled.div`
  ${({ theme, isVisible }) => css`
    height: 12rem;

    z-index: 1;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;

    border-bottom: 0.1rem solid ${theme.colors.gray};

    transition: all 300ms ease-in-out;

    & ${PageContainer} {
      min-height: initial;
      height: 100%;
    }

    @media (max-width: 1300px) {
      & ${PageContainer} {
        padding: 0 3rem;
      }
    }

    @media (max-width: 785px) {
      & {
        height: 100vh;
        bottom: 0;
        border-bottom: none;

        visibility: hidden;
        opacity: 0;
        ${
          isVisible &&
          css`
          visibility: visible;
          opacity: 1;
        `
        }
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${() => css`
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-end;

    @media (max-width: 785px) {
      & {
        height: 100%;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        gap: 7.5rem;
      }
    }
  `}
`;

export const Logo = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    align-items: flex-start;

    & ${Heading} {
      text-align: center;
    }

    @media (max-width: 1130px) {
      & ${Heading} {
        font-size: ${theme.fonts.fontSizes.headings.medium};
        margin-right: 2rem;
      }
    }

    @media ${theme.media.mediumDevices} {
      & ${Heading} {
        font-size: ${theme.fonts.fontSizes.headings.small};
      }
    }

    @media (max-width: 785px) {
      & {
        justify-content: center;
        align-items: center;
      }

      & ${Heading} {
        margin-right: 0;
      }
    }
  `}
`;

export const Nav = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    gap: 3rem;

    & ${Input} {
      width: 40.4rem;
    }

    @media (max-width: 1130px) {
      & {
        align-items: center;
      }

      & ${Input} {
        width: 30rem;
      }
    }

    @media ${theme.media.mediumDevices} {
      & ${Input} {
        width: 22.5rem;
      }
    }

    @media (max-width: 785px) {
      & {
        flex-direction: column;
        justify-content: center;
        align-items: center;
      }

      & ${InputContainer}, & ${Input}, & .dropdown-button  {
        display: none;
      }
    }
  `}
`;

export const Link = styled.a`
  ${({ theme }) => css`
    display: block;
    text-align: center;
    color: ${theme.colors.dark};

    font-weight: ${theme.fonts.fontWeights.light};
    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    line-height: 2.1rem;

    &:hover {
      cursor: pointer;
    }

    @media (max-width: 785px) {
      & {
        font-size: 1.8rem;
      }
    }
  `}
`;

export const Theme = styled.p`
  ${({ theme }) => css`
    display: none;
    color: ${theme.colors.dark};
    text-align: center;

    font-weight: ${theme.fonts.fontWeights.light};
    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    line-height: 2.1rem;

    &:hover {
      cursor: pointer;
    }

    @media (max-width: 785px) {
      & {
        display: block;
        font-size: 1.8rem;
      }
    }
  `}
`;
