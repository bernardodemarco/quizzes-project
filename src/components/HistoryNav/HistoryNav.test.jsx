import { fireEvent, screen } from '@testing-library/react';

import { renderThemeProvider } from './../../styles/render-theme-provider';

import { HistoryNav } from '.';

describe('<HistoryNav />', () => {
  it('should render <HistoryNav /> correctly', () => {
    const func = jest.fn();
    renderThemeProvider(<HistoryNav onReturnButtonClick={func} />);

    expect(
      screen.getByRole('heading', { name: 'Seu histÃ³rico' }),
    ).toBeInTheDocument();

    const nav = screen.getByRole('navigation');
    const returnButton = nav.firstChild;

    expect(nav).toBeInTheDocument();

    fireEvent.click(returnButton);
    expect(func).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    const func = jest.fn();
    const { container } = renderThemeProvider(
      <HistoryNav onReturnButtonClick={func} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
