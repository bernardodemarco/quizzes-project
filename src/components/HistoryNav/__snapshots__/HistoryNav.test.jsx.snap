// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoryNav /> should match snapshot 1`] = `
.c5 {
  width: 100%;
  max-width: 125rem;
  min-height: 100vh;
  margin: 0 auto;
  overflow-x: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  width: 4.8rem;
  height: 4.8rem;
  position: fixed;
  top: 3.6rem;
  left: 7.5rem;
  z-index: 2;
}

.c2:hover {
  cursor: pointer;
}

.c3 {
  width: 100%;
  height: 100%;
}

.c7 {
  color: #3C3A36;
  font-size: 4rem;
  font-weight: 700;
  line-height: 4.2rem;
}

.c0 {
  z-index: 1;
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  height: 12rem;
  border-bottom: 0.1rem solid #BEBAB3;
}

.c0 .c4 {
  min-height: initial;
  height: 100%;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .c6 {
  z-index: 1;
  position: fixed;
  left: 14.7rem;
}

@media (max-width:974px) {
  .c0 .c1 {
    left: 5rem;
  }
}

@media (max-width:575px) {
  .c0 .c6 {
    left: 12rem;
    font-size: 3.2rem;
  }
}

<nav
  class="c0"
>
  <div
    class="c1 c2"
  >
    <img
      alt="Return button"
      class="c3"
      src="assets/images/back-icon.svg"
    />
  </div>
  <div
    class="c4 c5"
  >
    <h1
      class="c6 c7"
      font-size="large"
      font-weight="bold"
    >
      Seu hist√≥rico
    </h1>
  </div>
</nav>
`;
